src/System.Configuration.ConfigurationManager/tests/System/Configuration/UrlPathTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Configuration;
+using System.IO;
+using Xunit;
+
+namespace System.ConfigurationTests
+{
+    public class UrlPathTests
+    {
+        [Fact]
+        public void GetDirectoryOrRootName_Null()
+        {
+            string test = UrlPath.GetDirectoryOrRootName(null);
+            Assert.Equal(null, null);
+        }
+
+        [Fact]
+        public void GetDirectoryOrRootName_NotDirectoryOrRoot()
+        {
+            string test = UrlPath.GetDirectoryOrRootName("Hello");
+            Assert.Equal("", test);
+        }
+
+        [Fact]
+        public void GetDirectoryOrRootName_GettingDIrectoryFromAFilePath()
+        {
+
+            string exePath = AppDomain.CurrentDomain.BaseDirectory;
+            //Remove the trailing slash.  Different OS's use a different slash.
+            //This is to make the test pass without worrying about adding a slash 
+            //and which kind of slash.
+            string exePathWithoutTrailingSlash = exePath.Substring(0, exePath.Length - 1);
+            string pathToNonexistentFile = exePath + "TestFileForUrlPathTests.txt";
+
+            string test = UrlPath.GetDirectoryOrRootName(pathToNonexistentFile);
+            Assert.Equal(exePathWithoutTrailingSlash, test);
+        }
+
+        [Fact]
+        public void GetDirectoryOrRootName_ofRoot()
```

Comment:
Please use "Of" instead of "of" here per .NET naming guidelines -- PascalCasing of method names.